src/python/nimbusml/ensemble/votingregressor.py

```diff
@@ -0,0 +1,64 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------------------------
+"""
+VotingRegressor
+"""
+
+__all__ = ["VotingRegressor"]
+
+
+from sklearn.base import RegressorMixin
+from ..base_predictor import BasePredictor
+from ..internal.utils.entrypoints import EntryPoint
+from ..internal.utils.utils import trace
+from ..internal.core.base_pipeline_item import DefaultSignature
+from ..internal.entrypoints.models_regressionensemble import \
+    models_regressionensemble
+
+
+class VotingRegressor(BasePredictor,
```

Comment:
I think `VotingRegressor` may be the wrong name. Voting is one option to combine model output for classification. One doesn't ensembing using _voting_ for regression, then tend to use {median, average, weighted average, stacking}, with options for model selection of {outlier removal, diversity, best-performance}.

I might suggest naming as `EnsembleRegressor`, or `RegressionEnsembler`:
```suggestion
class EnsembleRegressor(BasePredictor,
```

Reply:
Btw, if you make a stacked ensemble regression example/unit-test, I'd recommend a final model of LogisticRegression w/ non-negative set. The non-negative helps by ensuring you don't bet against the sub-models: https://github.com/dotnet/machinelearning/issues/1651#issuecomment-449535672

Reply:
The name `VotingRegressor` was chosen to match the class in `scikit-learn`. See [`VotingRegressor`](https://scikit-learn.org/stable/modules/ensemble.html#voting-regressor).

Though, everyone that has seen these changes has made similar comments about the name. Perhaps it would be best to change it.

There is already an `EnsembleRegressor` class in NimbusML (which is actually being updated to offer the same functionality as this class; this class is just a temporary class to add support for multi-predictor ensembling functionality until the existing class modifications are completed). `RegressionEnsembler` seems like it might be too similar to `EnsembleRegressor` and might lead to confusion.

@justinormont Thanks for your input. I will continue looking for other names.

Reply:
Was there a decision on naming?

Reply:
We have decided to keep the naming. Arguments are:
1. EnsembleRegressor already exists in NimbusML (where its restricted to the only base learner is Online Gradient Descent)
2. To be on par with scikit-learn where its named also VotingRegressor.

Once we are able to fix existing EnsembleRegressor to take any base learner we would obsolete this VotingRegressor in future.

---
In reply to: [370876925](https://github.com/microsoft/NimbusML/pull/390#discussion_r370876925) [](ancestors = 370876925)